81-IDevID_and_IAK_support.md

```diff
@@ -0,0 +1,313 @@
+
+#### Story 2
+<!-- TODO: add LDevID scenario with the user creating its own identity. -->
+With the IAK provided by the OEM, the client can issue it's own LDevID signed by it's own CAs. Instead of a long lived certificate as the IDevID from the OEM, the client can renew this certificate on a more frequent basis and securely verify its binding to the IAK and the devices TPM. 
+Along with the LDevID, it can create LAKs for different services or workloads, enabling a secure execution of services issuing short lived LAKs according to the demand in place.
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+enhancement ecosystem.
+
+How will security be reviewed and by whom?
+-->
+The risk involved at this proposal is increasing the configuration complexity and by enabling using different chains of trust for the IDs.
+
+For mitigation it is necessary to properly document what is trusted during registration and what is not, besides providing examples on how to configure it correctly.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+A potential workflow is presented below for the scenario using EK, EK certificate, IDevID, IAK, LDevID and LAK. The other scenarios will include most parts of this workflow. It also worth mentioning that when using an IAK the EK is not needed to enroll another AK (being either an EAK or an LAK). Instead of using make/activate credential, which requires a decryption key, TPM2_Certify is used to prove that the EAK/LAK (including the sensitive area, e.g., the plaintext private key) is co-resident in the same TPM as the IAK.
+
+### Registration from the agent perspective when interacting with the registrar service.
+1. Create session (done at get_tpm2_ctx)   
+2. Get endorsement certificate from TPM NV index and EK public key (done at create_ek)
+3. Regeneration of IAK and IDevID possible info (create create_iak and create_idevid routines). Depending on the configuration, the IDevID/IAK may also require a session.
+4. Load the IDevID and IAK and their certificates from a given local path and send them along with the rest of the attestation/measured data to the registrar service
+5. Create keys for LAK and LDevID based on the IAK (if possible, load them if already created). When using the TPM2_Certify the nonce should be used here.
```

Comment:
"load them" means allocate them in the NV of the TPM? Is this resource not scarce and recommended not to be used?

Reply:
Again, bear with my lack of knowledge (I take this opportunity to learn). If I{AK,DevID} are not present, is it possible to crate the L versions? If so, how the data for the DevID part is generated?